front/chat/src/Xmpp/MucRoom.ts

```diff
@@ -0,0 +1,318 @@
+                }
+            }
+        }
+
+        if (!handledMessage) {
+            console.warn("Unhandled message targeted at the room: ", xml.toString());
+            console.warn("Message name : ", xml.getName());
+        }
+    }
+
+    private getCurrentStatus(jid: string) {
+        return get(this.presenceStore).get(jid)?.status ?? USER_STATUS_DISCONNECTED;
+    }
+
+    private getCurrentRoomId(jid: string) {
+        return get(this.presenceStore).get(jid)?.roomId ?? '';
+    }
+
+    private getCurrentUuid(jid: string) {
+        return get(this.presenceStore).get(jid)?.uuid ?? '';
+    }
+
+    private getCurrentColor(jid: string) {
+        return get(this.presenceStore).get(jid)?.color ?? '#626262';
+    }
+
+    private getCurrentWoka(jid: string) {
+        return get(this.presenceStore).get(jid)?.woka ?? ' data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAdCAYAAABBsffGAAAB/ElEQVRIia1WMW7CQBC8EAoqFy74AD1FqNzkAUi09DROwwN4Ag+gMQ09dcQXXNHQIucBPAJFc2Iue+dd40QZycLc7c7N7d7u+cU9wXw+ryyL0+n00eU9tCZIOp1O/f/ZbBbmzuczX6uuRVTlIAYpCSeTScumaZqw0OVyURd47SIGaZ7n6s4wjmc0Grn7/e6yLFtcr9dPaaOGhcTEeDxu2dxut2hXUJ9ioKmW0IidMg6/NPmD1EmqtojTBWAvE26SW8r+YhfIu87zbyB5BiRerVYtikXxXuLRuK058HABMyz/AX8UHwXgV0NRaEXzDKzaw+EQCioo1yrsLfvyjwZrTvK0yp/xh/o+JwbFhFYgFRNqzGEIB1ZhH2INkXJZoShn2WNSgJRNS/qoYSHxer1+qkhChnC320ULRI1LEsNhv99HISBkLmhP/7L8OfqhiKC6SzEJtSTLHMkGFhK6XC79L89rmtC6rv0YfjXV9COPDwtVQxEc2ZflIu7R+WADQrkA7eCH5BdFwQRXQ8bKxXejeWFoYZGCQM7Yh7BAkcw0DEnEEPHhbjBPQfCDvwzlEINlWZq3OAiOx2O0KwAKU8gehXfzu2Wz2VQMTXqCeLZZSNvtVv20MFsu48gQpDvjuHYxE+ZHESBPSJ/x3sqBvhe0hc5vRXkfypBY4xGcc9+lcFxartG6LgAAAABJRU5ErkJggg==';
+    }
+
+    private updateUser(jid: string, nick: string, roomId: string|null = null, uuid: string|null = null, status: string|null = null, color: string|null = null, woka: string|null = null) {
+        const user = get(userStore);
+        if (
+            (MucRoom.encode(user?.email) ?? MucRoom.encode(user?.uuid)) + "@ejabberd" !== jid &&
+            nick !== this.getPlayerName()
+        ) {
+            this.presenceStore.update((list) => {
+                list.set(jid, {
+                    name: nick,
+                    roomId: roomId ?? this.getCurrentRoomId(jid),
+                    uuid: uuid ?? this.getCurrentUuid(jid),
+                    status: status ?? this.getCurrentStatus(jid),
+                    isInSameMap: (roomId ?? this.getCurrentRoomId(jid)) === user.playUri,
+                    active: (status ?? this.getCurrentStatus(jid)) === USER_STATUS_AVAILABLE,
+                    color: color ?? this.getCurrentColor(jid),
+                    woka: woka ?? this.getCurrentWoka(jid),
+                    unreads: false
+                });
+                numberPresenceUserStore.set(list.size);
+                return list;
+            });
+        }
+    }
+
+    public getPresenceStore(): UsersStore {
+        return {
+            subscribe: this.presenceStore.subscribe,
+        };
+    }
```

Comment:
Simply casting it as a UsersStore (which is a Readable) is enough.

```suggestion
    public getPresenceStore(): UsersStore {
        return this.presenceStore;
    }
```

Typescript will do the rest :D